src/pd/retry.rs

```diff
@@ -66,21 +69,47 @@ impl RetryClient<Cluster> {
             cluster,
             connection,
             timeout,
+            cache: RwLock::new(RegionCache::new()),
         })
     }
 
     // These get_* functions will try multiple times to make a request, reconnecting as necessary.
-    pub async fn get_region(self: Arc<Self>, key: Vec<u8>) -> Result<Region> {
+    pub async fn get_region(self: Arc<Self>, key: Key, skip_cache: bool) -> Result<Region> {
+        if skip_cache {
+            self.cache.write().unwrap().remove_region_by_start_key(&key);
```

Comment:
The `key` here isn't always the start key of a region. Maybe we can add a new method to the region cache.

Reply:
@nrc PTAL